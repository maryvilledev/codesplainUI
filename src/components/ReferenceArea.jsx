import React from 'react';

import mtIcon from '../../res/mt.png';
import twitterIcon from '../../res/twitter.png';
import githubIcon from '../../res/github-icon-blue.png';

const style = {
  paper: {
    padding: '0.7rem',
    display: 'inline-flex',
    flexFlow: 'row nowrap',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  linkArea: {
    display: 'inline-block',
    font: '12pt Lato, sans-serif',
  },
  logoArea: {
    display: 'block',
    textAlign: 'center',
  },
  link: {
    padding: '4px',
    color: '#666666',
  },
};

const ReferenceArea = () => (
  <div style={style.paper}>
    <a
      href="https://maryvilledevcenter.com/"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src={mtIcon}
        alt="Maryville DevCenter Logo"
        width={80}
      />
    </a>
    <div style={style.linkArea}>
      <div style={style.logoArea}>
        <a
          href="https://www.twitter.com/MaryvilleDev"
          target="_blank"
          rel="noopener noreferrer"
          style={style.link}
        >
          <img
            src={twitterIcon}
            alt="Twitter Icon"
            width={30}
          />
        </a>
        <a
          href="https://github.com/maryvilledev"
          target="_blank"
          rel="noopener noreferrer"
          style={style.link}
        >
          <img
            src={githubIcon}
            alt="Github Icon"
            width={30}
          />
        </a>
      </div>
      <a
        href="mailto:support@codesplain.io"
        target="_blank"
        rel="noopener noreferrer"
        style={style.link}
      >
          Contact Us
        </a>
    </div>
  </div>
);

export default ReferenceArea;
