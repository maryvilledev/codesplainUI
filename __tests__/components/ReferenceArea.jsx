import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ReferenceArea from '../../src/components/ReferenceArea';

describe('<ReferenceArea />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ReferenceArea />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
