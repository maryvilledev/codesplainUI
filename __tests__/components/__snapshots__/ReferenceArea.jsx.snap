exports[`<ReferenceArea /> matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
      "flexFlow": "row nowrap",
      "justifyContent": "flex-end",
      "padding": "0.7rem",
    }
  }>
  <a
    href="https://maryvilledevcenter.com/"
    rel="noopener noreferrer"
    target="_blank">
    <img
      alt="Maryville DevCenter Logo"
      src="mt.png"
      width={80} />
  </a>
  <div
    style={
      Object {
        "display": "inline-block",
        "font": "12pt Lato, sans-serif",
      }
    }>
    <div
      style={
        Object {
          "display": "block",
          "textAlign": "center",
        }
      }>
      <a
        href="https://www.twitter.com/MaryvilleDev"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#666666",
            "padding": "4px",
          }
        }
        target="_blank">
        <img
          alt="Twitter Icon"
          src="twitter.png"
          width={30} />
      </a>
      <a
        href="https://github.com/maryvilledev"
        rel="noopener noreferrer"
        style={
          Object {
            "color": "#666666",
            "padding": "4px",
          }
        }
        target="_blank">
        <img
          alt="Github Icon"
          src="github-icon-blue.png"
          width={30} />
      </a>
    </div>
    <a
      href="mailto:support@codesplain.io"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#666666",
          "padding": "4px",
        }
      }
      target="_blank">
      Contact Us
    </a>
  </div>
</div>
`;
